extract_region_feature.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+Region features extraction script.
+"""
+import argparse
+import os
+import torch
+import tqdm
+from fvcore.common.file_io import PathManager
+
+import numpy as np
+from torch.nn import functional as F
+
+from detectron2.checkpoint import DetectionCheckpointer
+from detectron2.config import get_cfg
+from detectron2.engine import default_setup
+from detectron2.evaluation import inference_context
+from detectron2.modeling import build_model
+from detectron2.modeling import postprocessing
+
+from grid_feats import (
+    add_attribute_config,
+    build_detection_test_loader_with_attributes,
+    build_detection_test_loader_for_images,
+)
+
+# A simple mapper from object detection dataset to VQA dataset names
+dataset_to_folder_mapper = {}
+dataset_to_folder_mapper["coco_2014_train"] = "train2014"
+dataset_to_folder_mapper["coco_2014_val"] = "val2014"
+# One may need to change the Detectron2 code to support coco_2015_test
+# insert "coco_2015_test": ("coco/test2015", "coco/annotations/image_info_test2015.json"),
+# at: https://github.com/facebookresearch/detectron2/blob/master/detectron2/data/datasets/builtin.py#L36
+dataset_to_folder_mapper["coco_2015_test"] = "test2015"
+
+
+def extract_feature_argument_parser():
+    parser = argparse.ArgumentParser(description="Region feature extraction")
+    parser.add_argument(
+        "--config-file", default="", metavar="FILE", help="path to config file"
+    )
+    parser.add_argument(
+        "--dataset",
+        help="name of the dataset",
+        default="coco_2014_train",
+        choices=["coco_2014_train", "coco_2014_val", "coco_2015_test"],
+    )
+    parser.add_argument(
+        "--dataset-path",
+        help="path to image folder dataset, if not provided expects a detection dataset",
+        default="",
+    )
+    parser.add_argument(
```

Comment:
maybe worth adding an option of "K" which stands for extracting the top K region features, I think this is useful for a lot of papers that choose top 36, or varying size from 36-100 (maybe another option is needed there)